File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int pos = nums.length;
        for(int i=0;i<nums.length-1;i++){
               if(nums[i]==target) pos=i;
        else if(nums[i]< target && target<=nums[i+1])
                pos= i+1;
        }
        
       if(target<=nums[0])
                pos=0;
        
        
    return pos;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        
        left = 0
        right = len(nums) #target could be bigger than the last value in the array
        
        while left < right:
            mid = left + (right - left) // 2
            if target == nums[mid]:
                return mid
            elif target <= nums[mid]:
                right = mid
            else:
                left = mid + 1
        return left
        